site/docs/hosting/fly.md

```diff
@@ -0,0 +1,426 @@
+# Hosting: Fly <Badge text="Deno" /><Badge text="Node.js" type="warning"/>
+
+This guide tells you about the ways you can host your grammY bots on [Fly](https://fly.io), either using Deno or Node.js.
+
+## Preparing Your Code
+
+You can run your bot either on [webhooks](../guide/deployment-types.md) or [long polling](../guide/deployment-types.md).
+
+### Webhooks
+
+> Remember that you should not call `bot.start()` in your code when using webhooks.
+
+You can use the example code provided in our [Deno Deploy](./deno-deploy.md#preparing-your-code) (Deno) and [Heroku](./heroku.md#express-and-its-middleware) (Node.js) hosting guide.
+In this section, we literally transcribe the text from the both docs.
+
+1. Make sure that you have a file which exports your `Bot` object, so that you can import it later to run it.
+2. Create a file named `app.ts` or `app.js`, or actually any name you like (but you should be remembering and using this as the main file to deploy), with the following content:
+
+<CodeGroup>
+<CodeGroupItem title="Deno" active>
+
+```ts{6,11}
```

Comment:
Why are these two lines highlighted? This should either be explained (“as shown in the first highlighted line”), or the highlighting should be removed.